SeQuant/core/utility/singleton.hpp

```diff
@@ -96,16 +96,16 @@ class Singleton {
           "sequant::Singleton::set_instance: instance has already been "
           "constructed");
     instance_accessor() = std::move(
-        std::shared_ptr<Derived>(new Derived(std::forward<Args>(args)...)));
-    return instance_accessor();
+        std::unique_ptr<Derived>(new Derived(std::forward<Args>(args)...)));
```

Comment:
`move` should not be needed here.
Also, we probably should make use of `std::make_unique`